node/mock-task.ts

```diff
@@ -407,11 +407,13 @@ export function matchFile(list, pattern, options): string[] {
     return mock.getResponse('match', pattern) || [];
 }
 
-export function filter(pattern, options): any {
-    var filterList = mock.getResponse('filter', pattern) || [];
-	return function(pattern, i, arr) {
-		return filterList.indexOf(pattern) >= 0;
-	}
+export function filter(pattern, options): (element: string, index: number, array: string[]) => boolean {
```

Comment:
With Moq mocking (C#) it's possible to mock some calls to a method (based on param values), and have others fall through to the base implementation (if `CallBase=true`), so that part doesn't feel odd to me.

From my test author's point of view the base implementation of 'filter' is perfectly good - generate a function that matches values on a pattern - there's no external dependencies on the environment.

My motivation was that I had a portion of build task code `tl.find(path).filter(tl.filter('*.exe'))`, and I wanted to mock `find` (since I didn't want to rely on the real file system). The current mocking framework forces me to _also_ get involved in mocking `filter` (whose default behavior would have been perfectly good), which then makes my test code unnecessarily messy.

I'd be happy not mocking `filter` at all, but if I do want to, then I don't really want to be forced to mock _every_ occurrence that appears _somewhere_ in the task code. I'd prefer to be targetted with the mocking.